File: src/p0011/golang/solution.go
package main

func maxArea(h []int) int {
	result, i, j := 0, 0, len(h) - 1
	for i < j {
		if h[i] < h[j] {
			if result < (j - i) * h[i] {
				result = (j - i) * h[i]
			}
			i++
		} else {
			if result < (j - i) * h[j] {
				result = (j - i) * h[j]
			}
			j--
		}
	}
	return result
}

File: src/p0011/python/solution.py
from typing import List


class Solution:
    def maxArea(self, h: List[int]) -> int:
        result, i, j = 0, 0, len(h) - 1
        while i < j:
            if h[i] < h[j]:
                result, i = max(result, (j - i) * h[i]), i + 1
            else:
                result, j = max(result, (j - i) * h[j]), j - 1
        return result
